shared/mail/mail.go

```diff
@@ -307,6 +316,14 @@ func sendMail(c smtpClient, mail mailData, date time.Time, config *SMTPConfig) e
 		"Precedence":                {"bulk"},
 	}
 
+	if len(mail.category) > 0 {
+		sendgridHeader, headerErr := json.Marshal(XSMTPAPIHeader{Category: mail.category})
+		if headerErr != nil {
+			return errors.Wrapf(headerErr, "Failed to marshal X-SMTPAPI header")
+		}
```

Comment:
I think we might be overcomplicating this with marshalling to json, and then casting from `[]byte` to `string` again. I'd just go ahead with 

```
fmt.Sprintf(`{"category": %q}`, mail.category)
```

0/5